src/reason_parser.mly

```diff
@@ -2705,6 +2707,13 @@ _simple_expr:
         let loc = mklocation $symbolstartpos $endpos in
         bigarray_get ~loc $1 $4
       }
+  | simple_expr VARIABLEINFIXOP simple_expr
```

Comment:
what's your grammar rule there?

Try add this under "expr as(loc(infix_operator) expr":
```
  | expr VARIABLEINFIXOP expr
      {
        let n = prepare_name_for_call $2 in
        let attribute = simple_ghost_text_attr "infix" in
        let name = mkexp ~ghost:true ~attrs:(attribute) (Pexp_ident(ghloc (Longident.parse n))) in
        mkexp (Pexp_apply (name, ["", $1; "", $3]))
      }
```
Notice we have "mark_position_exp" after expr so you don't need to calculate ~loc and pass is as an argument to mkexp.